united.py

```diff
@@ -28,10 +54,224 @@ def parse_date(created):
+
+
+    ]
+
+
+def convert_xml_numbers_to_current_weather(weather_code):
+    print weather_code
+    str_weather = str(weather_code)
+    str_weather = str_weather.strip(' ')
+    weather = {"0": 1, "1": 2, "3": 3, "4": 4, "5": 5, "7": 6, "8": 6, "9": 7, "10": 8, "11": 9,
+               "12": 10, "17": 11, "18": 12, "19": 13, "20": 14, "21": 15, "22": 16, "23": 17, "24": 18,
+               "25": 19, "26": 20, "27": 21, "28": 22, "29": 23, "30": 24, "31": 24, "32": 24, "33": 7,
+               "34": 7, "35": 7, "36": 25, "37": 25, "38": 25, "39": 25, "40": 26, "41": 27, "42": 28,
+               "43": 29, "44": 9, "45": 30, "46": 30, "47": 30, "48": 31, "49": 32, "50": 33, "51": 34,
+               "52": 33, "53": 35, "54": 36, "55": 37, "56": 38, "57": 39, "58": 37, "59": 37, "61": 37, "60": 36,
+               "62": 40, "63": 15, "64": 41, "65": 19, "66": 42, "67": 43, "68": 44, "69": 45, "70": 46, "71": 47,
+               "72": 48, "73": 16, "74": 50, "75": 51, "76": 52, "77": 53, "78": 54, "79": 55, "80": 56, "81": 57,
+               "82": 58, "83": 59, "84": 60, "85": 61, "86": 62, "87": 63, "88": 64, "89": 65, "90": 66, "91": 67,
+               "92": 68, "93": 69, "94": 70, "95": 71, "96": 72, "97": 73, "98": 74, "99": 75}
+
+    return weather[str_weather]
+
+
+def process_weather_data(latitude, longitude):
+
+    weather = 1  # default weather is clear sky
+    rain_in_millimeters = 0
+    rain_hours = 0
+
+    station = find_station_by_coordinate(latitude, longitude)
+    weather_data = collection.getElementsByTagName('surface_observation')
+
+    wind_force = weather_data[station].getElementsByTagName('FF')
+    for wf in wind_force:
+        wind_force = wf.childNodes[0].nodeValue
+
+    rain = weather_data[station].getElementsByTagName('RRR')
+    for r in rain:
+        rain = r.childNodes[0].nodeValue
+
+    rain_duration = weather_data[station].getElementsByTagName(
+        'TR')  # the duration of time in which  the rain amount was measured
+    for rd in rain_duration:
+        rain_duration = rd.childNodes[0].nodeValue
+
+    weather_code = weather_data[station].getElementsByTagName('WW')
+    for wc in weather_code:
+        weather_code = wc.childNodes[0].nodeValue
+
+    if weather_code:
+        weather = convert_xml_numbers_to_current_weather(weather_code)
+
+    else:
+        if rain:
+            rain_in_millimeters = convert_xml_values_to_numbers(rain)
+        if rain_duration:
+            rain_hours = convert_xml_numbers_to_hours(rain_duration)
+
+        if wind_force:
+
+            if int(wind_force) > 8:
+                weather = 76  # סופת רוחות
+            elif int(wind_force) > 5:
+                weather = 77  # רוחות חזקות
+
+        if rain:
+            if rain_duration:
+                # rain amount is between 0.1 and 0.5 millimeter
+                if rain_in_millimeters > 0.0 and rain_in_millimeters <= 0.5 or (rain_in_millimeters/rain_hours > 0.0 and rain_in_millimeters/rain_hours <= 0.5):
```

Comment:
Use combined comparison here and whenever applicable:
`0.0 < rain_in_millimeters <= 0.5`